text/0045-add-notion-of-protocol.md

```diff
@@ -0,0 +1,116 @@
+# Summary
+- Add notion of service protocol in libobs and OBS Studio
+- Outputs for streaming are registered with their compatible protocol and codecs
+- Only codecs compatible with the output and the service are shown
+- Audio encoder can be choosen if various compatible
+- Will be extended with RFC 39
+
+# Motivation
+Create a better management of outputs, encoders and services with an approach focused on protocols.
+
+# Design
+
+## Outputs API
+Only outputs with the `OBS_OUTPUT_SERVICE` flag are concerned.
+
+Adding to `obs_output_info`:
+- `const char *protocols`: protocols separated with a semi-colon (ex: `"RTMP;RTMPS"`), required to register the output.
+- `const char *protocols_prefixes`: URL protocols prefixes separated with a semi-colon `"rtmp://;rtmps://"`, optional.
```

Comment:
I require the `://` to avoid:
- matching with links (without schemes) like `rtmp.example.com`.
- adding it each time that a "auto-detection" based on it.

I may need to explain it in the RFC.

Reply:
This is probably fine as-is.